From 8dabdde5608fb43430a3d0ec6a2d402f81f5bbd7 Mon Sep 17 00:00:00 2001
Message-Id: <8dabdde5608fb43430a3d0ec6a2d402f81f5bbd7.1228954349.git.christian@jaeger.mine.nu>
In-Reply-To: <030d49bbacadc03ef604478be7d0582bd2703cf7.1228954349.git.christian@jaeger.mine.nu>
References: <030d49bbacadc03ef604478be7d0582bd2703cf7.1228954349.git.christian@jaeger.mine.nu>
From: Christian Jaeger <christian@jaeger.mine.nu>
Date: Wed, 10 Dec 2008 23:55:54 +0100
Subject: [PATCH] Fix attempt: search in kernel/drivers/ too

But as long as mod is 'dm_mod' and the kernel (2.6.27.7 in my case)
provides dm-mod, it still won't be found.

Signed-off-by: Christian Jaeger <christian@jaeger.mine.nu>
---
 debian/initramfs/cryptroot-hook |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/debian/initramfs/cryptroot-hook b/debian/initramfs/cryptroot-hook
index f8b197e..3243ebd 100644
--- a/debian/initramfs/cryptroot-hook
+++ b/debian/initramfs/cryptroot-hook
@@ -413,6 +413,15 @@ add_crypto_modules() {
 		found="yes"
 	done
 
+	if [ -z "$found" ]; then
+		for file in $(find "$MODULESDIR/kernel/drivers/" -name "$mod-*.ko"); do
+			altmod="${file##*/}"
+			altmod="${altmod%.ko}"
+			manual_add_modules "$altmod"
+			found="yes"
+		done
+	fi
+
 	for file in $(find "$MODULESDIR/kernel/crypto/" -name "${mod}_generic.ko"); do
 		altmod="${file##*/}"
 		altmod="${altmod%.ko}"
-- 
1.6.0.5

